packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/method_channel_cloud_firestore.dart

```diff
@@ -0,0 +1,356 @@
+    bool isCollectionGroup,
+    Map<String, dynamic> parameters,
+    bool includeMetadataChanges,
+  }) {
+    // Create a stream of query snapshot handles, as they happen
+        assert(includeMetadataChanges != null);
+    Future<int> _handle;
+    // It's fine to let the StreamController be garbage collected once all the
+    // subscribers have cancelled; this analyzer warning is safe to ignore.
+    StreamController<int> controller; // ignore: close_sinks
+    controller = StreamController<int>.broadcast(
+      onListen: () {
+        _handle = _addQuerySnapshotListener(app, path: path, isCollectionGroup: isCollectionGroup, parameters: parameters, includeMetadataChanges: includeMetadataChanges,)
+        .then<int>((dynamic result) => result);
+        _handle.then((int handle) {
+          _queryObservers[handle] = controller;
+        });
+      },
+      onCancel: () {
+        _handle.then((int handle) async {
+          await removeListener(handle);
+          _queryObservers.remove(handle);
+        });
+      },
+    );
+    return controller.stream;
+  }
+
+  // Broadcast the QuerySnapshot data
+  void _handleQuerySnapshot(MethodCall call) {
+    final int handle = call.arguments['handle'];
+    _queryObservers[handle].add(call.arguments);
+  }
+
+  //TODO: Type this return
+  @override
+  Future<Map<dynamic, dynamic>> getQueryDocuments(String app, {
+    @required String path,
+    bool isCollectionGroup,
+    Map<String, dynamic> parameters,
+    String source,
+  }) => channel.invokeMapMethod<String, dynamic>(
+      'Query#getDocuments',
+      <String, dynamic>{
+        'app': app,
+        'path': path,
+        'isCollectionGroup': isCollectionGroup,
+        'parameters': parameters,
+        'source': source,
+      },
+    );
+
+  // Transaction
+  // TODO: Type this return
+  @override
+  Future<Map<String, dynamic>> getTransaction(String app, {
+    @required String path,
+    @required int transactionId,
+  }) => channel
+        .invokeMapMethod<String, dynamic>('Transaction#get', <String, dynamic>{
+      'app': app,
+      'transactionId': transactionId,
+      'path': path,
+    });
+
+  @override
+  Future<void> deleteTransaction(String app, {
+    @required String path,
+    @required int transactionId,
+  }) => channel
+        .invokeMethod<void>('Transaction#delete', <String, dynamic>{
+      'app': app,
+      'transactionId': transactionId,
+      'path': path,
+    });
+
+  @override
+  Future<void> updateTransaction(String app, {
+    @required String path,
+    @required int transactionId,
+    Map<String, dynamic> data,
+  }) => channel
+        .invokeMethod<void>('Transaction#update', <String, dynamic>{
+      'app': app,
+      'transactionId': transactionId,
+      'path': path,
+      'data': data,
+    });
+
+  @override
+  Future<void> setTransaction(String app, {
+    @required String path,
+    @required int transactionId,
+    Map<String, dynamic> data,
+  }) => channel
```

Comment:
Only use `=>` when everything, including the function declaration, fits on a single line.